### ISSUE TITLE
Implementation for distributed batch normalisation 

### ISSUE THREAD

**Issue by banditboi** [pytorch/pytorch#2844]
Hello,
I want to implement distributed batch normalisation for multiple GPUs. 
If someone already implemented this for Pytorch, I would be greatful if you can share the code with me.
I am also looking for various ideas on how to implement multi GPU batchnorm. 

My idea is to maintain different statistics for each respective GPU and randomly select one GPU statistic during test time. Is this a good idea?

**Comment by apaszke** [comment]
Maybe it is, maybe it isn't. I don't think we'll be including this in the core unless there's a paper that shows that it's worth it.
